src/components/API.js

```diff
@@ -0,0 +1,178 @@
+export default class API {
+  constructor(options) {
+    this._baseUrl = options.baseUrl;
+    this._headers = options.headers;
+  }
+
+  getInitialCards() {
+    return fetch(`${this._baseUrl}/cards`, {
+      method: "GET",
+      headers: this._headers,
+    })
+      .then((response) => {
+        if (response.ok) {
+          return response.json();
+        }
+        return Promise.reject("Error:", error);
+      })
+      .catch((error) => console.error("Error:", error));
```

Comment:
You should **not** use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request, and this end is in `index.js` after all `then` blocks. And in real projects the `catch` blocks usually open error popups and so on.